### PR TITLE
Add bpf2c_plugin for use with bpf_conformance test suite

### PR DESCRIPTION
**Pull request by Alan-Jowett**
## Description

Add a new test binary, bpf2c_plugin, that can be used with the bpf_conformance test suite to verify that bpf2c conforms to the BPF ISA.

## Testing

New tests.

## Documentation

No.

### CODE REVIEW

File: tests/bpf2c_plugin/bpf2c_plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,208 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+// This program reads BPF instructions from stdin and memory contents from
+// the first agument. It then executes the BPF program and prints the
```
Comment (dthaler) [inline tests/bpf2c_plugin/bpf2c_plugin.cpp:5]:
```suggestion
// the first argument. It then executes the BPF program and prints the
```

File: tests/bpf2c_plugin/bpf2c_test.cpp [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+metadata_table_t*
+get_metadata_table()
+{
+    return &metadata_table;
+}
+
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#pragma once
```
Comment (dthaler) [inline tests/bpf2c_plugin/bpf2c_test.cpp:35]: Why is this here in the middle of the cpp file?  Delete?

Comment (Alan-Jowett) [reply tests/bpf2c_plugin/bpf2c_test.cpp]: This was a header that I cut and pasted into the template to make things simpler. Fixed now.

**Review by dthaler — APPROVED** [review]

File: tests/bpf2c_plugin/bpf2c_test.cpp [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+        for (size_t j = 0; j < helper_function_entry_count; j++) {
+            if (helper_function_entries[j].helper_id == -1) {
+                std::cout << "bpf_test doesn't support resolving helpers by name yet." << std::endl;
+                return -1;
+            }
+            if (_helper_functions.find(helper_function_entries[j].helper_id) == _helper_functions.end()) {
+                std::cout << "bpf_test doesn't support helper id=" << helper_function_entries[j].helper_id << std::endl;
+                return -1;
+            } else {
```
Comment (gtrevi) [inline tests/bpf2c_plugin/bpf2c_test.cpp:168]: nit: the else statement could be removed (always executed after the two "check-out" tests fail), makes its section more distinct with less nesting.

File: docs/InstallEbpf.md [hunk 4]
```diff
@@ -54,9 +54,9 @@ to install eBPF on all Windows nodes in a Kubernetes cluster.
 
 1. Deploy the binaries to `C:\Temp` on the machine (Windows Host) where you built the binaries.
    Start an admin Powershell on the Windows Host and do `.\x64\debug\deploy-ebpf`.
-   
-2. Build ebpf-for-windows image. 
-     
+
+2. Build ebpf-for-windows image.
```
Comment (gtrevi) [inline docs/InstallEbpf.md:58]: nit: markup allows numbered list items to be all "1.", which will be automatically enumerated upon rendering (makes it easier to maintain when needing to add/remove items without renumbering - exception is when cross-referencing between bullets is needed).

**Review by gtrevi — APPROVED** [review]

File: tests/bpf2c_plugin/bpf2c_test.cpp (hunk 3 above)
Comment (Alan-Jowett) [reply tests/bpf2c_plugin/bpf2c_test.cpp:168]: Oops, missed your comments. Will fix it in a subsequent PR.

File: docs/InstallEbpf.md (hunk 4 above)
Comment (Alan-Jowett) [reply docs/InstallEbpf.md:58]: Oops, missed your comments. Will fix it in a subsequent PR.
